File: Count the reversals/count reversal.java
class Sol
{
    int countRev (String s)
    {
        // your code here  
        if(s.length()%2!=0) return -1;
        Stack<Character> stack = new Stack<>();
        for(int i=0;i<s.length();i++){
            if(!stack.isEmpty() && s.charAt(i) == '}' && stack.peek()== '{'){
                stack.pop();
            } else
            stack.push(s.charAt(i));
        }
        int count=0;
        
        while(!stack.isEmpty()){
          char j = stack.pop();
          char i = stack.pop();
          if(j!='}') count++;
          if(i!='{') count++;
        }
        return count;
    }
}

File: Count the reversals/count reversal.cpp
void solve()
 {
    string str ;
      cin >> str ;
      if(str.size()%2!=0){
          cout<<"-1"<<endl;
      }
      else{
        stack<char> s;
        int unbalance =0,count =0;
        for(int i = 0 ; i<str.size() ;i++)
          {
            if(str[i] == '{')
                s.push(i);
                else
                {
                    if(s.empty())
                    unbalance++;
                    else
                    s.pop();  
                }
          }
                cout<<(unbalance +1)/2 +((s.size()+1)/2)<<endl;  
        }
}
        
